lib/grid_dao.es6.js

```diff
@@ -18,18 +18,67 @@ foam.CLASS({
   ],
 });
 
+
+foam.CLASS({
+  package: 'org.chromium.mlang.sink',
+  name: 'Projection',
+  extends: 'foam.mlang.ExprProperty',
+
+  documentation: `Projection over particular data columns. Primarliy used to
+      implement GridDAO.projectCols().
+
+      E.g., myGridDAO.select(MAP(myGridDAO.projectCols(
+                myGridDAO.cols[10],
+                myGridDAO.cols[7],
+                myGridDAO.cols[45])))`,
+
+  properties: [
+    {
+      class: 'Array',
+      of: 'Int',
+      name: 'cols',
+    },
+    {
+      name: 'f',
+      value: function f(gridRow) {
+        let row = gridRow.clone();
```

Comment:
It might be better to only copy the elements that are needed, and then construct a row that wraps the new array.

Reply:
Interesting point. The only trade-off I see there is that if `GridRow` is later changed to have more properties, then this code needs to be updated to stay in sync.

Reply:
That is the downside. If you have enough data in the grid, there's probably a measurable cost to copying the whole thing and then GCing the discarded portion of it.

Reply:
Created a new entity copying only ID (then set data when it's ready). In case of a modeled ID, I attempt to clone the ID (and otherwise use assignment).